Neos.Fusion/Tests/Unit/FusionObjects/DataStructureImplementationTest.php

```diff
@@ -133,4 +142,34 @@ public function evaluateRendersKeysSortedByPositionMetaProperty($message, $subEl
 
         self::assertSame($expectedKeyOrder, $renderedPaths, $message);
     }
+
+    /**
+     * @test
+     * @dataProvider positionalSubElementsThatShouldFailByInvalidPositions
+     *
+     * @param string $message
+     * @param array $subElements
+     * @param array $expectedKeyOrder
+     */
+    public function evaluateRendersKeysSortedByPositionMetaPropertyThatShouldFail($message, $subElements, $expectedKeyOrder)
+    {
+        try {
+            $mockRuntime = $this->getMockBuilder(Runtime::class)->disableOriginalConstructor()->getMock();
+
+            $mockRuntime->expects(self::any())->method('evaluate')->will(self::returnCallback(function ($path) use (&$renderedPaths) {
+                $renderedPaths[] = $path;
+            }));
+
+            $path = '';
+            $fusionObjectName = 'Neos.Fusion:DataStructure';
+            $renderer = new DataStructureImplementation($mockRuntime, $path, $fusionObjectName);
+            foreach ($subElements as $key => $value) {
+                $renderer[$key] = $value;
+            }
+            $renderer->evaluate();
+            self::fail('Expected InvalidPositionException exception not thrown');
+        } catch (\Exception $exception) {
+            self::assertEquals($exception->getCode(), 1345126502);
+        }
```

Comment:
Any reason why you didn't use `$this->expectException()`?

Reply:
```$this->expectException(InvalidPositionException::class);```

Ends in another error for me that the expected exception could not be matched with a fusion exception :)

Reply:
There's also `$this->expectExceptionCode()`.
I adjusted this and added some (mostly cosmetic) tweaks. Will merge if CI is green